Проектная работа №2 Обобщение классов

nsuhover opened the pull request:
Сделано обобщение класса ListDiffCallback через новый дженерик-класс BaseListDiffCallback.

В дженерик-классе реализованы методы: areItemsTheSame, areContentsTheSame, getChangePayload

Создан интерфейс ListDiffInterface для ограничения использования типов данных в классах ContactUi, FilterContactTypeUi, SortTypeUI в методах theSameAs и equals

nsuhover committed:
В адаптерах ContactAdapter, FilterContactTypeAdapter, SortTypeAdapter сделано обобщение класса ListDiffCallback через новый дженерик-класс BaseListDiffCallback.
В дженерик-классе реализованы методы: areItemsTheSame, areContentsTheSame, getChangePayload

Создан интерфейс ListDiffInterface для ограничения использования типов данных в классах ContactUi, FilterContactTypeUi, SortTypeUI в методах theSameAs и equals

nsuhover committed:
исправлена ошибка в ContactUi, указан был не тот тип.

gusar-off commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/filter/model/FilterContactTypeUi.java:
```diff
@@ -20,6 +22,12 @@ public boolean isSelected() {
+
+    @Override
+    public boolean theSameAs(FilterContactTypeUi item) {
+        return this.hashCode() == item.hashCode();
```
Для сравнения старого и нового айтема в локальном колбеке `ListDiffCallback` использовалось 
```
return oldItem.getContactType() == newItem.getContactType();
```
для сравнения `FilterContactTypeUi`
В нашей обобщенной реализации это не должно измениться

nsuhover replied:
Илья, спасибо, поправил

gusar-off commented on app/src/main/java/ru/yandex/practicum/contacts/presentation/sort/SortTypeUI.java:
```diff
@@ -22,6 +23,12 @@ public boolean isSelected() {
+
+    @Override
+    public boolean theSameAs(SortTypeUI item) {
+        return this.hashCode() == item.hashCode();
```
И тут тоже было
```
return oldItem.getSortType() == newItem.getSortType();
```

nsuhover replied:
тут тоже, отправил новую ссылку

nsuhover committed:
Исправлено замечание:
"Неправильно реализован метод theSameAs() в FilterContactTypeUI, SortTypeUI"

FilterContactTypeUI в методе theSameAs использован метод getContactType.
SortTypeUI в методе theSameAs использован метод getSortType.
